### PR TITLE
executor: show create table ignores 'binary' collate table option

### LINKED ISSUE #15633: collate: SQL from show-create-table with collate binary parses error
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. What did you do?
As vividly illustrated as below, show create table result will contain the collate content.
but when SQL with the collate binary will be parsed error in return, it's wired! 
```
mysql> create table t(a int) charset=binary;
Query OK, 0 rows affected (0.15 sec)

mysql> show create table t;
+-------+--------------------------------------------------------------------------------------+
| Table | Create Table                                                                         |
+-------+--------------------------------------------------------------------------------------+
| t     | CREATE TABLE `t` (
  `a` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=binary COLLATE=binary |
+-------+--------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> drop table t;
Query OK, 0 rows affected (0.15 sec)

mysql> CREATE TABLE `t` (`a` int(11) DEFAULT NULL) ENGINE=InnoDB DEFAULT CHARSET=binary COLLATE=binary;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'binary' at line 1
```

### 2. What did you expect to see?
MySQL won't show the collate in the show-create-table result.
Since TiDB has already shown the collate in the result, we need to handle this!

1：parser the collate=binary in the parser
2：ignore the collate element in the show-create-table result when collate is binary.

### 3. What did you see instead?
show-create-result should execute successfully.

### 4. What version of TiDB are you using? (`tidb-server -V` or run `select tidb_version();` on TiDB)
4.0/3.1/3.0/2.1 all them

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #15633 

Related parser PR: pingcap/parser#789

Problem Summary:
The statements with `binary` collate table option like 
```sql
create table t (a varchar(10)) collate=binary
``` 
are unsupported syntactically in MySQL 5.7 and old version TiDB.

The result of `show create table` should ignore the `binary` collate table option.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### CODE REVIEW

**Comment by wjhuang2016** [comment]
Where is the parser PR?

**Comment by tangenta** [comment]
> Where is the parser PR?

Description updated. pingcap/parser#789

File: executor/show.go [hunk 1]
```diff
@@ -877,7 +877,7 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
 	buf.WriteString(") ENGINE=InnoDB")
 	// Because we only support case sensitive utf8_bin collate, we need to explicitly set the default charset and collation
```
Comment (wjhuang2016) [inline executor/show.go]: This comment is oudated since we supported new collation. Please refine it.

File: executor/show.go [hunk 2]
```diff
@@ -877,7 +877,7 @@ func ConstructResultOfShowCreateTable(ctx sessionctx.Context, tableInfo *model.T
...
 	// to make it work on MySQL server which has default collate utf8_general_ci.
-	if len(tblCollate) == 0 {
+	if len(tblCollate) == 0 || tblCollate == "binary" {
```
Comment (wjhuang2016) [inline executor/show.go]: Please add the comment for the reason.

**Comment by wjhuang2016** [comment]
Please update the PR

**Comment by tangenta** [comment]
This PR is blocked by #15412.

**Comment by tangenta** [comment]
@wjhuang2016 PTAL

**Pull request merged**